docs/source/tutorial/kernel.rst

```diff
@@ -263,16 +263,128 @@ Accessing texture (surface) memory in :class:`~cupy.RawKernel` is supported via
+| uint8           | unsigned char                                 | 1                |
++-----------------+-----------------------------------------------+------------------+
+| uint16          | unsigned short                                | 2                |
++-----------------+-----------------------------------------------+------------------+
+| uint32          | unsigned int                                  | 4                |
++-----------------+-----------------------------------------------+------------------+
+| uint64          | unsigned long long                            | 8                |
++-----------------+-----------------------------------------------+------------------+
+| float16         | half                                          | 2                |
++-----------------+-----------------------------------------------+------------------+
+| float32         | float                                         | 4                |
++-----------------+-----------------------------------------------+------------------+
+| float64         | double                                        | 8                |
++-----------------+-----------------------------------------------+------------------+
+| complex64       | float2, cuFloatComplex, complex<float>        | 8                |
++-----------------+-----------------------------------------------+------------------+
+| complex128      | double2, cuDoubleComplex, complex<double>     | 16               |
++-----------------+-----------------------------------------------+------------------+
+
+The CUDA standard guarantees that the size of fundamental types on the host and device always match.
+The itemsize of ``size_t``, ``ptrdiff_t``, ``intptr_t``, ``uintptr_t``, 
+``long``, ``signed long`` and ``unsigned long`` are thus platform dependent. 
+You can also pass any CUDA vector builtins such as ``float3`` or any other user defined structure 
+as kernel arguments provided it matches device side kernel parameter type, see section :ref:`custom_user_structs`.
+
+.. note::
+    To use ``cuFloatComplex`` and ``cuDoubleComplex`` in your CUDA kernel, you need to include the header ``cuComplex.h``. You may also need to pass `translate_cucomplex=True` to :class:`~cupy.RawKernel` or :class:`~cupy.RawModule`.
+
+.. note::
+    To use ``complex<float>`` and ``complex<double>`` in your CUDA kernel, you simply need to include the header ``<cupy/complex.cuh>``. This is the recommended way to get complex support in :class:`~cupy.RawKernel`.
+
+.. note::
+    To be able to directly use ``std::complex<float>`` and ``std::complex<double>``, you need to include the header ``<cuda/std/complex.h>`` which is provided by `NVIDIA C++ Standard Library` version 1.4.0 and above. Currently **this header is not shipped with CUDA toolkit 11.2**, see this `link <https://nvidia.github.io/libcudacxx/releases.html>`_ to get more information.
```

Comment:
IMHO I would suggest to remove this part for the following reasons:
1. CuPy currently has not included any integration test with libcu++, and while we are keen to the `std::complex` support (due to https://github.com/cupy/cupy/issues/3370) it is better to not recommend this usage until we have a good test coverage. 
2. This is oversimplified: Users not only have to include the header, but also have to set the include path via `options=('-I/path/to/libcu++/header', ...)` as libcu++ is not available in any standard (CUDA or CuPy) path, and perhaps some linker options as well (that I don't know).
3. libcu++'s `std::complex` support was buggy on NVRTC, which was just fixed very recently (https://github.com/NVIDIA/libcudacxx/pull/109), available since 1.4.1. I

In short, I think the current status of libcu++ is more for adventurous users, not average ones. For the former group they don't count on our tutorial here, and for the latter they'd just get confused.